staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -2414,6 +2414,7 @@ type ArchConfiguration struct {
 	Amd64               *ArchSpecificConfiguration `json:"amd64,omitempty"`
 	Arm64               *ArchSpecificConfiguration `json:"arm64,omitempty"`
 	Ppc64le             *ArchSpecificConfiguration `json:"ppc64le,omitempty"`
+	S390x               *ArchSpecificConfiguration `json:"s390x,omitempty"`
```

Comment:
Hi @xpivarc ,
git tells me it's been about 6 months since i touched that piece of code, but i think i got some error deploying a VM on s390x without it.
Or would that not be the case and this piece is just needed for emulation of s390x on other arches?

Reply:
I will spend some time debugging that tomorrow or friday and share my results.
But if it is only for running s390x VMs on other arches than yes it is not relevant for this PR.

Reply:
Ok, i have compiled a version without api changes, but even during build i already noticed what my problem was in the past:
```
{"component":"virt-launcher","level":"error","msg":"unsupported configuration: Emulator '/usr/libexec/qemu-kvm' does not support machine type 'q35'","pos":"qemuValidateDomainDef:1227","subcomponent":"libvirt","thread":"25","timestamp":"2024-01-18T12:06:37.798000Z"}
```
The Problem is that the generated struct is used to set the defaults for specific arches, including the machine type. Without that code, the default is always to use the amd64 default configuration, which of course does not work on s390x.
See: https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-config/configuration.go#L250C1-L267C5 and https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-config/virt-config.go#L134C1-L147C2

It might be possible to hardcode the returns in the necessary functions here if we need to avoid an api change with all cost, but i don't really thing it would be a good idea to diverge from the established pattern here.

Reply:
The `ArchConfiguration` is really meant only for multi-arch cluster support. It is ok to modify appropriate places to work for s390x, this was also done for arm.

>See: https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-config/configuration.go#L250C1-L267C5 and https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-config/virt-config.go#L134C1-L147C2

The former set the default to `runtime.GOARCH`, should it be s390x? The latter can be extended but without the need to define multi-cluster config.

Reply:
Well `runtime.GOARCH` resolves to `s390x` anyway on s390x.
But i can try to create a version where the values in the functions are hardcoded.

Reply:
Starting a basic, native VM works without the API changes.

Reply:
@xpivarc I'm probably missing something, but what is the advantage of leaving this out for now? If other architectures have it and we might want it for s390x too in the future, why not introduce it right away instead of starting with a crippled implementation?